src/instrument/mongoose_instrument_exometer.erl

```diff
@@ -160,15 +164,27 @@ update_metric(Name, histogram, Value) when is_integer(Value) ->
 -spec exometer_metric_name(mongoose_instrument:event_name(), mongoose_instrument:labels(),
                            mongoose_instrument:metric_name()) -> exometer:name().
 exometer_metric_name(EventName, Labels, MetricName) ->
-    [get_host_type_prefix(Labels), EventName] ++ exometer_labels(Labels) ++ [MetricName].
+    HostPrefix = get_host_type_prefix(Labels),
+    RemainingLabels = maps:without([host_type], Labels),
+    ExometerLabels = exometer_labels(RemainingLabels),
+    [HostPrefix, EventName] ++ ExometerLabels  ++ [MetricName].
 
 %% This logic will need extending if we add more labels
-exometer_labels(#{function := Function}) ->
+exometer_labels(#{function := Function} = Labels) when map_size(Labels) =:= 1 ->
     [Function];
-exometer_labels(#{pool_tag := PoolTag}) ->
+exometer_labels(#{pool_tag := PoolTag} = Labels) when map_size(Labels) =:= 1 ->
     [PoolTag];
-exometer_labels(#{}) ->
-    [].
+exometer_labels(#{pool_id := PoolId} = Labels) when map_size(Labels) =:= 1 ->
+    [PoolId];
+exometer_labels(#{connection_type := ConnectionType} = Labels) when map_size(Labels) =:= 1 ->
+    [ConnectionType];
+exometer_labels(#{cache_name := CacheName} = Labels) when map_size(Labels) =:= 1 ->
+    [CacheName];
+exometer_labels(#{} = Labels) when map_size(Labels) =:= 0 ->
+    [];
+exometer_labels(#{} = Labels) ->
```

Comment:
We might end up with unexpected warnings that we ignore (because we don't check warnings in tests)... that's my only concern. Wouldn't it be better to make it fail fast, i.e. omit this clause? We could also enforce correct label syntax in `mongoose_instrument` itself.